src/ProjectConfiguration.ts

```diff
@@ -617,6 +624,50 @@ export function validateCompanionDefaultWakeInterval(
   return diagnostics;
 }
 
+export function validateTileBuildTarget( tile: Tile, { buildTargets }: AppProjectConfiguration) {
+  if (tile.buildTargets !== undefined){
+    const diagnostic = constrainedSetDiagnostics({
+      actualValues: tile.buildTargets,
+      knownValues: buildTargets,
+      valueTypeNoun: 'tile build targets',
+      notFoundIsFatal: true
+    })
+    return diagnostic;
+  }
+
+  // Return empty list. Nothing to check
+  return new DiagnosticList();
+}
+
+export function validateTileUUID(tile: Tile, config: AppProjectConfiguration) {
+  const diagnostic = new DiagnosticList()
+  if (tile.uuid === undefined) {
+    diagnostic.pushFatalError("Tile UUID can't be undefined");
+  }
+
+  if (!validator.isUUID(String(tile.uuid))) {
+    diagnostic.pushFatalError(
+      'Tile UUID must be a valid UUID',
+    );
+  }
+
+  const knownUUID = [config.appUUID]
```

Comment:
Perhaps simplify these two lines to:

```
const knownUUIDs = [config.appUUID, ...config.tiles];
```

Reply:
Not really to expand this way as object have different types. Refactored to use `map` instead. Should be a bit cleaner now.